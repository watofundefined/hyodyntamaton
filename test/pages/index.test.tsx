import React from 'react'
import { render } from '../test-utils'
import Home from '../../pages/index'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders Log in button if user is logged out', () => {
    const { getByText } = render(<Home />)
    expect(() => getByText(/log in/i)).not.toThrow()
  })

  it('renders Welcome back when user is logged in', () => {
    const { asFragment } = render(<Home />, {
      initState: { user: { loggedIn: true, token: null } },
    })
    expect(asFragment()).toMatchSnapshot()
  })
})
