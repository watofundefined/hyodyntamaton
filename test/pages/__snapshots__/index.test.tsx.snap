// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="page-container"
  >
    <header>
      <span
        class="logo"
      >
        Hyödyntämätön
      </span>
    </header>
    <main>
      App which translates beer reviews to your language.
      <button>
        Log in with your Untappd account
      </button>
    </main>
  </div>
</DocumentFragment>
`;

exports[`Home page matches snapshot when user is logged in 1`] = `
<DocumentFragment>
  <div
    class="page-container"
  >
    Welcome back!
    <button>
      Show map
    </button>
    <button>
      Log out
    </button>
  </div>
</DocumentFragment>
`;

exports[`Home page renders Welcome back when user is logged in 1`] = `
<DocumentFragment>
  <div
    class="page-container"
  >
    Welcome back!
    <button>
      Show map
    </button>
    <button>
      Log out
    </button>
  </div>
</DocumentFragment>
`;
